### PR TITLE
tasks

### PR DESCRIPTION
**Pull request by goose-life**

### CODE REVIEW

**Comment by goose-life** [comment]
see #300

File: indigo_app/templates/indigo_api/task_detail.html [hunk 1]
```diff
@@ -20,6 +20,42 @@
...
+        <p>{% user_profile task.last_closed_by_user %} closed this task <span class="time-ago" data-timestamp="{{ task.last_closed_at|date:'c' }}">{{ task.last_closed_at|naturaltime }}</span>
+</p>
+        {% endif %}
+
+      {% elif reopened_and_unsubmitted %}
+      <p>
+      {% for change in reopened_and_unsubmitted|dictsort:'at' reversed %}
+        {% if forloop.first %}
+          {% user_profile change.user %} {{ change.change }} this task
+          {% if change.change == 'reopened' %}
```
[the remark this answers is not in the document]
Comment (goose-life) [reply indigo_app/templates/indigo_api/task_detail.html]: @longhotsummer yep it's to only show the most recent change. 
the other states are real states so we can just look that up on the task and display the current state and who put it in that state (if it's not just `open`), 
but 'reopened' and 'unsubmitted' are pseudostates (they're both 'open'), so I thought to see if the task had a `last_reopened_by` or `last_unsubimtted_by` recorded, but I only want the most recent if it's had both done to it. 
I definitely want to make this part less clunky but I don't understand your advice to replace the if block with one that uses `change.at`?

[the remark this answers is not in the document]
Comment (goose-life) [reply indigo_app/templates/indigo_api/task_detail.html]: I had it displaying both so I could double check it was behaving as expected before, but yes it definitely isn't necessary anymore if I only want the most recent

File: indigo_app/views/tasks.py [hunk 2]
```diff
@@ -42,6 +42,39 @@ class TaskDetailView(TaskViewBase, DetailView):
...
+                 'at': task.last_reopened_at},
+                {'change': 'unsubmitted',
+                 'user': task.last_unsubmitted_by_user,
+                 'at': task.last_unsubmitted_at}
+            ]
+
+        if self.request.user.has_perm('indigo_api.change_task'):
+            context['change_task_permission'] = True
+
+        if has_transition_perm(task.submit, self):
```
Comment (longhotsummer) [inline indigo_app/views/tasks.py]: You could also put these on one line each:

```python
context['submit_task_permission'] = has_transition_perm(task.submit, self)
```

**Pull request merged**